src/modules/aws-ebs/sub-modules/snapshots/checkSnapshotProgress.ts

```diff
@@ -0,0 +1,31 @@
+import { EC2 } from "aws-sdk";
+
+type SnapshotProgress = {
+  state: string;
+  progress: string;
+  volumeSize: number;
+};
+
+export const checkSnapshotProgress = async (SnapshotId: string): Promise<SnapshotProgress> => {
+  const ec2 = new EC2();
+  const params = {
+    SnapshotIds: [SnapshotId],
+  };
+  const snapshotAttribute = await ec2.describeSnapshots(params).promise();
+
+  if (
+    !snapshotAttribute.Snapshots ||
```

Comment:
/nit could probably yank this into a var representing the same array value that you're referencing.
i.e. 
```
const snapshot = snapshotAttribute.Snapshots[0];
```

Reply:
Not sure if I can do that without typescript complaining about "Object is possibly 'undefined'.  For this one, I was doing short circuit evaluation which is why typescript is not complaining.

Perhaps, I could do something like: 
```
  const snapshotAttribute = await ec2.describeSnapshots(params).promise();

  if (!snapshotAttribute.Snapshots || !snapshotAttribute.Snapshots[0]) {
    throw new Error(`Snapshot ${SnapshotId} is missing or has missing required attributes.`);
  }

  const snapshot = snapshotAttribute.Snapshots[0];
  if (!snapshot.State || !snapshot.Progress || !snapshot.VolumeSize) {
    throw new Error(`Snapshot ${SnapshotId} is missing or has missing required attributes.`);
  }
```

But I'm not sure if it makes it any better